app/controllers/transcriptions_controller.rb

```diff
@@ -114,6 +134,21 @@ def approving?
     update_attrs["status"] == "approved"
   end
 
+  def fresh?
+    # the 'If-Unmodified-Since' datetime will be sent over by client with ISO 8601 format, 3 digits of fractional seconds
+    @transcription.updated_at.iso8601(3) == request.headers['If-Unmodified-Since']
+  end
+
+  def locked?
+    @transcription.lock_timeout &&
+      DateTime.now < @transcription.lock_timeout &&
+      current_user.login != @transcription.locked_by
+  end
+
+  def lock_transcription
+    @transcription.update!(locked_by: current_user.login, lock_timeout: DateTime.now + 3.hours)
```

Comment:
This method should be moved to the model

Reply:
That makes sense. Apparently I have a tendency to put things in the controller if it's functionality that I need to access within a controller action, but I think i should start asking myself whether a method actually belongs on the model.